localstack/services/s3/provider.py

```diff
@@ -572,6 +587,67 @@ def delete_bucket_website(
         # does not raise error if the bucket did not have a config, will simply return
         self.get_store().bucket_website_configuration.pop(bucket, None)
 
+    def post_object(
+        self, context: RequestContext, bucket: BucketName, body: IO[Body] = None
+    ) -> PostResponse:
+        # see https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html
+        # TODO: signature validation is not implemented for pre-signed POST
+        # policy validation is not implemented either, except expiration and mandatory fields
+        validate_post_policy(context.request.form)
+
+        # Botocore has trouble parsing responses with status code in the 3XX range, it interprets them as exception
+        # it then raises a nonsense one with a wrong code
+        # We have to create and populate the response manually if that happens
+        try:
+            response: PostResponse = call_moto(context=context)
+        except ServiceException as e:
+            if e.code == "303":
```

Comment:
nit: should probably be
```suggestion
            if e.status_code == 303:
```